src/taxonomy.rs

```diff
@@ -158,19 +133,15 @@ impl<'t, T, D> TaxonomyIterator<'t, T, D> {
     }
 }
 
-impl<'t, T, D> Iterator for TaxonomyIterator<'t, T, D>
-where
-    T: Clone + Debug + Display + PartialEq,
-    D: Debug + PartialOrd + PartialEq + Sum,
-{
-    type Item = (T, bool);
+impl<'t> Iterator for TaxonomyIterator<'t> {
+    type Item = (&'t str, bool);
```

Comment:
It might be helpful to have a comment here explaining what the bool is. I know elsewhere it says it's to indicate preorder / postorder traversal, but I can't quite tell the purpose of it. But again I'm quite unfamiliar with this crate, so maybe it's obvious to those using it.

Reply:
Honestly I'm not 100% sure